kornia/core/module.py

```diff
@@ -0,0 +1,273 @@
+                    # Convert all kwargs to tensors
+                    kwargs = {k: self.to_tensor(v) if self._is_valid_arg(v) else v for k, v in kwargs.items()}
+                else:
+                    # Convert specified args to tensors
+                    args = list(args)
+                    for i, (arg, name) in enumerate(zip(args, func.__code__.co_varnames)):
+                        if name in input_names_to_handle:
+                            args[i] = self.to_tensor(arg)
+                    # Convert specified kwargs to tensors
+                    for name, value in kwargs.items():
+                        if name in input_names_to_handle:
+                            kwargs[name] = self.to_tensor(value)
+
+                # Call the actual forward method
+                tensor_outputs = func(*args, **kwargs)
+
+                if not isinstance(tensor_outputs, tuple):
+                    tensor_outputs = (tensor_outputs,)
+
+                # Convert outputs to the desired type
+                outputs = []
+                for output in tensor_outputs:
+                    if output_type == "tensor":
+                        outputs.append(output)
+                    elif output_type == "numpy":
+                        outputs.append(self.to_numpy(output))
+                    elif output_type == "pil":
+                        outputs.append(self.to_pil(output))
+                    else:
+                        raise ValueError("Output type not supported. Choose from 'tensor', 'numpy', or 'pil'.")
+
+                return outputs if len(outputs) > 1 else outputs[0]
+
+            return wrapper
+
+        return decorator
+
+    def _is_valid_arg(self, arg: Any) -> bool:
+        """Check if the argument is a valid type for conversion.
+
+        Args:
+            arg: The argument to check.
+
+        Returns:
+            bool: True if valid, False otherwise.
+        """
+        if isinstance(arg, (str,)) and os.path.exists(arg):
+            return True
+        if isinstance(arg, (Tensor,)):
+            return True
+        # Make sure that the numpy and PIL are not necessarily needed to be imported.
+        if isinstance(arg, (np.ndarray,)):
+            return True
+        if isinstance(arg, (Image.Image)):
+            return True
+        return False
+
+    def to_tensor(self, x: Any) -> Tensor:
+        """Convert input to tensor.
+
+        Supports image path, numpy array, PIL image, and raw tensor.
+
+        Args:
+            x: The input to convert.
+
+        Returns:
+            Tensor: The converted tensor.
+        """
+        if isinstance(x, str):
+            return kornia.io.load_image(x, kornia.io.ImageLoadType.UNCHANGED) / 255
+        elif isinstance(x, Tensor):
+            return x
+        elif isinstance(x, np.ndarray) and len(x.shape) == 3:
+            return from_numpy(x).permute(2, 0, 1).float() / 255
```

Comment:
reuse `image_to_tensor` -- to avoid circular maybe lets move utils.image under kornia.core.image